pkg/admission/server.go

```diff
@@ -144,45 +145,33 @@ func handleValidation(request admission.AdmissionRequest) (*admission.AdmissionR
 	switch request.Resource {
 	case v1a1httpRouteGVR:
 		var hRoute v1alpha1.HTTPRoute
-		deserializer := codecs.UniversalDeserializer()
 		_, _, err := deserializer.Decode(request.Object.Raw, nil, &hRoute)
 		if err != nil {
 			return nil, err
 		}
 
-		fieldErr := v1a1Validation.ValidateHTTPRoute(&hRoute)
-		if fieldErr != nil {
-			message = fmt.Sprintf("%s", fieldErr.ToAggregate())
-			ok = false
-		} else {
-			ok = true
-		}
+		fieldErr = v1a1Validation.ValidateHTTPRoute(&hRoute)
 	case v1a2httpRouteGVR:
 		var hRoute v1alpha2.HTTPRoute
-		deserializer := codecs.UniversalDeserializer()
 		_, _, err := deserializer.Decode(request.Object.Raw, nil, &hRoute)
 		if err != nil {
 			return nil, err
 		}
-
-		fieldErr := v1a2Validation.ValidateHTTPRoute(&hRoute)
-		if fieldErr != nil {
-			message = fmt.Sprintf("%s", fieldErr.ToAggregate())
-			ok = false
-		} else {
-			ok = true
-		}
+		fieldErr = v1a2Validation.ValidateHTTPRoute(&hRoute)
 	default:
 		return nil, fmt.Errorf("unknown resource '%v'", request.Resource.Resource)
 	}
 
 	response.UID = request.UID
-	response.Allowed = ok
-	response.Result = &meta.Status{
-		Message: message,
-	}
-	if !ok {
-		response.Result.Code = 400
+	if len(fieldErr) != 0 {
+		response.Allowed = false
+		response.Result = &meta.Status{
+			Message: fmt.Sprintf("%s", fieldErr.ToAggregate()),
+			Code:    400,
+		}
+	} else {
+		response.Allowed = true
+		response.Result = &meta.Status{}
```

Comment:
Nit, I think it would be marginally clearer to return `AdmissionResponse` literals:

```Go
if len(fieldError) > 0 {
    return admission.AdmissionResponse {
        ...
    }
}

return admission.AdmissionResponse {
    UID: request.UID,
    Allowed: true,
}
```
```